### PR TITLE
Fix builds on riscv64 using musl

### LINKED ISSUE #25772: musl riscv64 crossbuild failure with openssl since v3.4.0
Musl version is git master https://git.musl-libc.org/cgit/musl/commit/

With the release of v3.4.0 I saw this error when cross building openssl v3.4.0

```
riscv64-linux-musl-gcc  -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include  -DAES_ASM -DGHASH_ASM -DINCLUDE_C_CHACHA20 -DINCLUDE_C_SHA256 -DINCLUDE_C_SHA512 -DOPENSSL_CPUID_OBJ -DOPENSSL_SM3_ASM -DSHA256_ASM -DSHA512_ASM -DSM4_ASM -DSTATIC_LEGACY -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSLDIR="\"/etc/ssl\"" -DENGINESDIR="\"/root/qbt-build/lib/engines-3\"" -DMODULESDIR="\"/root/qbt-build/lib/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG -static -w -Wno-psabi -I/root/qbt-build/include -MMD -MF crypto/libcrypto-lib-self_test_core.d.tmp -c -o crypto/libcrypto-lib-self_test_core.o crypto/self_test_core.c
crypto/riscvcap.c: In function 'riscv_hwprobe':
crypto/riscvcap.c:85:20: error: '__NR_riscv_hwprobe' undeclared (first use in this function); did you mean 'riscv_hwprobe'?
   85 |     return syscall(__NR_riscv_hwprobe, pairs, pair_count, cpu_count, cpus, flags);
      |                    ^~~~~~~~~~~~~~~~~~
      |                    riscv_hwprobe
crypto/riscvcap.c:85:20: note: each undeclared identifier is reported only once for each function it appears in
make[1]: *** [Makefile:9690: crypto/libcrypto-lib-riscvcap.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/root/qbt-build/openssl-3.4.0'
make: *** [Makefile:2936: build_sw] Error 2
```

possibly related to this?

https://github.com/openssl/openssl/commit/66ad636b979554ddde5cd5908feabda79d07317b

### PR DESCRIPTION
**Pull request by mattcaswell**
Some environments using musl are reported to have the `hwprobe.h` include file but not have the `__NR_riscv_hwprobe` define.

Fixes #25772

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
Ping @openssl/committers for second review.

**Review by kaduk — APPROVED** [review]

**Review by vdukhovni — APPROVED** [review]

**Comment by tmshort** [comment]
Merged to `master` and `openssl-3.4`. Thank you for your contribution!

**Pull request closed without merging**
